lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,283 @@
+import userEvent from '@testing-library/user-event';
+
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+
+import NxDrawer, { Props } from '../NxDrawer';
+import NxButton from '../../NxButton/NxButton';
+import useToggle from '../../../util/useToggle';
+
+describe('NxDrawer', function() {
+  const minimalProps: Props = {
+    open: true,
+    onClose: () => {},
+    children: 'Drawer Content'
+  };
+
+  const quickRender = rtlRender<React.ComponentPropsWithRef<typeof NxDrawer>>(NxDrawer, minimalProps);
+  const getDrawer = rtlRenderElement(NxDrawer, minimalProps);
+
+  it('renders <dialog> with class nx-drawer', function () {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const dialog = getDrawer()!;
+    expect(dialog.nodeName).toBe('DIALOG');
+    expect(dialog).toHaveClass('nx-drawer');
+  });
+
+  it('has a dialog element with aria-modal set to false', function() {
+    const dialog = getDrawer();
+    expect(dialog).toHaveAttribute('aria-modal', 'false');
+  });
+
+  it('it does not have nx-drawer--closing class when initially loaded', function () {
+    const dialog = getDrawer();
+    expect(dialog).not.toHaveClass('nx-drawer--closing');
+  });
+
+  it('renders children nodes within dialog', function() {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const drawer = getDrawer({ children: <div data-testid="foo"/> })!;
+    expect(within(drawer).getByTestId('foo')).toBe(drawer.children[0]);
+  });
+
+  it('merges any passed in className to the nx-drawer dialog', function() {
+    const drawerWithAddedClassName = getDrawer({ className: 'foo' });
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const drawer = getDrawer()!;
+
+    expect(drawerWithAddedClassName).toHaveClass('foo');
+
+    for (const cls of Array.from(drawer.classList)) {
+      expect(drawerWithAddedClassName).toHaveClass(cls);
+    }
+  });
+
+  it('includes any passed in attributes to the dialog element', function() {
+    const drawer = getDrawer({ id: 'drawer-id', lang: 'en_US' });
+    expect(drawer).toHaveAttribute('id', 'drawer-id');
+    expect(drawer).toHaveAttribute('lang', 'en_US');
+  });
+
+  it('adds the nx-drawer--narrow class when the narrow variant prop is specified', function() {
+    const drawer = getDrawer({ variant: 'narrow' });
+    expect(drawer).toHaveClass('nx-drawer--narrow');
+  });
+
+  describe('NxDrawer event listener support', () => {
+    it('executes onClose callback when ESC key is pressed', function() {
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      getDrawer({ onClose: mockOnClose })!;
+
+      const dialog = screen.getByRole('dialog', { hidden: true });
+      expect(mockOnClose).not.toHaveBeenCalled();
+
+      act(() => {
+        fireEvent.keyDown(dialog, { key: 'Escape' });
```

Comment:
Why are we using `act` and `fireEvent` here but `userEvent` in the next test?

Reply:
> Why are we using `act` and `fireEvent` here but `userEvent` in the next test?

Cleaned up here: 3abd242de90de93f44f84124fcd18ff81db5f1a5